src/python_testing/TC_HSTAT_2_3.py

```
@@ -0,0 +1,215 @@
+        # Verify: If no reports were received, fail the test.
+        # If (MaxSetpointValue - MinSetpointValue) = StepValue, verify 1 report was received and the value is MaxSetpointValue.
+        # If (MaxSetpointValue - MinSetpointValue) > StepValue, verify 2 reports were received and: The value for the first report is MaxSetpointValue.
+        # The value for the second report is MinSetpointValue + StepValue.
+        dut_Setpoint = await self.read_attribute_expect_success(attribute=self.attributes.UserSetpoint)
+        asserts.assert_equal(dut_Setpoint, dut_MinSetpoint+dut_Step, "UserSetpoint is not MinSetpoint+Step as expacted")
+        setpoint_span = dut_MaxSetpoint - dut_MinSetpoint
+        expected_report_count = 1 if setpoint_span == dut_Step else 2
+        asserts.assert_equal(len(reportsReceived), expected_report_count, "Unexpected report count")
+        asserts.assert_equal(reportsReceived[0], dut_MaxSetpoint, "First report value is not MaxSetpoint")
+        if expected_report_count == 2:
+             asserts.assert_equal(reportsReceived[1], dut_MinSetpoint+dut_Step,
```

Flake8 (7.3.0)

[error] indentation is not a multiple of 4

(E111)